Source/RN4UE4/ReplicaRigidDynamicClient.cpp

```diff
@@ -40,19 +167,70 @@ RigidDynamicConstructionData UReplicaRigidDynamicClient::GetConstructionData()
 
 	FMatrix conversionMatrix = FMatrix();
 	memcpy(conversionMatrix.M, matrixElements, 16 * sizeof(float));
-
+	actorTransform.SetScale3D(FVector(-actorTransform.GetScale3D().X, actorTransform.GetScale3D().Y, actorTransform.GetScale3D().Z));
 	actorTransform *= FTransform(conversionMatrix.Inverse());
 	actorTransform.ScaleTranslation(1 / 50.0f);
 
-	RigidDynamicConstructionData data;
-	data.geom = 0;
-	data.pos.X = -actorTransform.GetLocation().X;
+	GetNearestStaticMesh();
+	ReadPhysicValues(data);
+
+	FTransform centerMassTransform = FTransform(FQuat(data.centerMassRot.X, data.centerMassRot.Y, data.centerMassRot.Z, data.centerMassRot.W), FVector(data.centerMass.X, data.centerMass.Y, data.centerMass.Z), FVector(-1.0f, 1.0f, 1.0f));
+	centerMassTransform *= FTransform(conversionMatrix.Inverse());
+	centerMassTransform.ScaleTranslation(1 / 50.0f);
+
+	data.clientCreated = true;
+	data.pos.X = actorTransform.GetLocation().X;
 	data.pos.Y = actorTransform.GetLocation().Y;
 	data.pos.Z = actorTransform.GetLocation().Z;
 	data.rot.X = actorTransform.GetRotation().X;
 	data.rot.Y = actorTransform.GetRotation().Y;
 	data.rot.Z = actorTransform.GetRotation().Z;
 	data.rot.W = actorTransform.GetRotation().W;
+	data.scale.X = actorTransform.GetScale3D().X;
+	data.scale.Y = actorTransform.GetScale3D().Y;
+	data.scale.Z = actorTransform.GetScale3D().Z;
+	data.numVertex = 0;
+
+	if (orionMesh->GetBodySetup()->AggGeom.BoxElems.Num() > 0) {
+		data.geom = 3;
+		data.scale.X = orionMesh->GetBodySetup()->AggGeom.BoxElems[0].X / 2.0f / 50.0f * data.scale.X;
+		data.scale.Y = orionMesh->GetBodySetup()->AggGeom.BoxElems[0].Z / 2.0f / 50.0f * data.scale.Z;
+		data.scale.Z = orionMesh->GetBodySetup()->AggGeom.BoxElems[0].Y / 2.0f / 50.0f * data.scale.Y;
+	}
+	else if (orionMesh->GetBodySetup()->AggGeom.SphylElems.Num() > 0) {
+		data.geom = 2;
+		data.scale.Y = orionMesh->GetBodySetup()->AggGeom.SphylElems[0].Length / 2.0f / 50.0f * data.scale.Z;
+		data.scale.X = orionMesh->GetBodySetup()->AggGeom.SphylElems[0].Radius / 50.0f * data.scale.X;
+	}
+	else if (orionMesh->GetBodySetup()->AggGeom.SphereElems.Num() > 0) {
+		data.geom = 0;
+		data.scale.X = orionMesh->GetBodySetup()->AggGeom.SphereElems[0].Radius / 50.0f * data.scale.X;
+	}
+	//if mesh
+	else if (orionMesh->GetBodySetup()->AggGeom.ConvexElems.Num() > 0) {
+		data.geom = 4;
+		data.numVertex = orionMesh->GetBodySetup()->AggGeom.ConvexElems[0].VertexData.Num();
+		for (FVector vec : orionMesh->GetBodySetup()->AggGeom.ConvexElems[0].VertexData)
+		{
+			FVector aux;
+			aux.X = vec.X - data.centerMass.X;
```

Comment:
Again with the individual component assignment and `centerMass`/`centerMassRot` below